### PR TITLE
Clarify stepsize and discourage usage of CS for real sampled data systems but recommend HCS or SES instead

### LINKED ISSUE #434: Clarification of stepSize for CoSim FMUs
**Reported by maplesoft on 1 Jun 2018 17:21 UTC**
The meaning of stepSize, if provided in the DefaultExperiment field, needs to be clarified.
On pg. 43 of the spec it says:
"for CoSimulation the _stepSize_ defines the **preferred** communicationStepSize"
but on pg. 96 it says:
"the **smallest** sample period as accessible by the outside of the FMU is defined by attribute _stepSize_ in element DefaultExperiment"

So should the importing tool interpret it as a minimum or just a preferred step size?  This is important especially when canHandleVariableCommunicationStepSize = false.

----

Migrated-From: https://trac.fmi-standard.org/ticket/434

### PR DESCRIPTION
**Pull request by chrbertsch**
attempt to fix #434

### CODE REVIEW

**Comment by pmai** [comment]
Is the wording not a tad too strong? I.e. do we have reasons to suggest that the use of CS is problematic for all sampled data systems?

**Comment by chrbertsch** [comment]
> Is the wording not a tad too strong? I.e. do we have reasons to suggest that the use of CS is problematic for all sampled data systems?

Should we have something like "... FMI for CS can be used for simple sampled data systems that can be modeled with certain assumptions like
- no events
- one timestep delay
- no defintion of various "tasks" ..."
for which the new features of HCS (events, clocks) or SES shall be used.

**Comment by andreas-junghanns** [comment]
Even if such an exclusion list would be helpful, it is misleading. "No events" is not true, one can have events, just the communication is more difficult and maybe besides the standard. And a time step is not defined, really, as one can reduce the time step and remove the delay, if so needed, under certain assumptions.

I think it is better to explain what the new simulation types are good for. That would be simpler to explain, I think. Let´s make this a brief discussion tomorrow.

File: docs/4_1_co-simulation_math.adoc [hunk 1]
```diff
@@ -81,15 +81,12 @@ This section contains a formal mathematical model of a Co-Simulation FMU.
...
-
-- A hybrid ODE that is integrated between communication points (known as "sampled access to time continuous systems") where internal events may occur and be handled, but events are not visible from the outside of the FMU.
+Such a sampled-data system consists typically of a hybrid ODE that is integrated between communication points (known as "sampled access to time continuous systems") where internal events may occur and be handled, but events are not visible from the outside of the FMU.
 It is assumed here that all <<input,`inputs`>> and all <<output,`outputs`>> of this hybrid ODE are Real signals (defined with <<variability>> = <<continuous>>),
 
-- A combination of the systems above.
+_[ In FMI 2.0, Co-Simulation FMUs were also used for real sampled-data systems (so a sampled discrete controller; the <<input,`inputs`>> and <<output,`outputs`>> could be of type Real, Integer, Boolean, String, or enumeration with <<variability>> = <<discrete>>._
+_However, in FMI 3.0, Hybrid Co-Simulation (HCS) and Scheduled Execution Simulation (SES) are intended for this use-case.]_
```
Comment (APillekeit) [inline docs/4_1_co-simulation_math.adoc]: Frankly, I don't understand the point of this discussion. Would it be helpful if we change inline 89 "are intended" into "are more suitable".

Comment (APillekeit) [reply docs/4_1_co-simulation_math.adoc]: We explain the use cases of the different interface types at another place. 
(see pull request https://github.com/modelica/fmi-standard/pull/728)
I would not repeat that here. The issue #434 is solved in my view with the updates Christian provided.

**Pull request merged**